resolver/resolver.go

```diff
@@ -85,30 +81,34 @@ func (res *Resolver) Serve(net string) {
 
 	err := server.ListenAndServe()
 	if err != nil {
-		logging.Error.Printf("Failed to setup "+net+" server: %s\n", err.Error())
+		return fmt.Errorf("Failed to setup %q server: %v", net, err)
 	} else {
 		logging.Error.Printf("Not listening/serving any more requests.")
 	}
-
-	os.Exit(1)
+	return nil
 }
 
-// launches Zookeeper detector
-func (res *Resolver) LaunchZK() {
-	dr, err := res.ZKdetect()
-	if err != nil {
-		logging.Error.Println(err.Error())
-		os.Exit(1)
-	}
+// launches Zookeeper detector, returns immediately
+func (res *Resolver) LaunchZK() <-chan error {
+	errCh := make(chan error, 1)
+	go func() {
```

Comment:
@jdef There is one issue now that we lunch ZKdetect as a goroutine. The first reload() of state happens before ZKdetect gets the initial master and one has to wait for a reload cycle for Mesos-DNS to work. This is annoying for testing (not a major issue after that though). 

Is this go routine really needed?